rasa/nlu/training_data/message.py

```diff
@@ -24,16 +26,16 @@
 class Message:
     def __init__(
         self,
-        text: Text,
+        text: Optional[Text] = "",
         data: Optional[Dict[Text, Any]] = None,
         output_properties: Optional[Set] = None,
         time: Optional[Text] = None,
         features: Optional[List["Features"]] = None,
         **kwargs,
     ) -> None:
-        self.text = text
         self.time = time
-        self.data = data if data else {}
+        self.data = data.copy() if data else {}
+        self.data.update({TEXT: text})
```

Comment:
why do we need that?

Reply:
For line 38: Partially, because of the `init` problem I described above. And because we need to make sure all the passed input is stored in the data. 
For line 37: because `text` as a definitive property is gone. and now imagine the case where I want to initialise 2 messages as follows:
```
data = {intent: my_intent}
texts = ['hello', 'hi']
message1 = Message(texts[0], data)
message2 = Message(texts[1], data)
```
then I want to change the intent in message1:
`message1.data["intent"] = "another_intent"`
 (if we don't make a copy) this will change the data of message2 as well because they came from the same `data` variable initially.

Reply:
let's update the places with passing `data` instead of text during initialization

Reply:
Done in #6394